podvm/qcow2/misc-settings.sh

```diff
@@ -38,4 +38,14 @@ then
 # else if...
 #
 fi
+
+
+if [ -e /etc/certificates/tls.crt ] && [ -e /etc/certificates/tls.key ] && [ -e /etc/certificates/ca.crt ]; then
+    # Update systemd service file to add additional options
+    sed -i "/ExecStart/s/$/ \-cert\-file \/etc\/certificates\/tls.crt \-cert\-key \/etc\/certificates\/tls.key \-ca\-cert\-file \/etc\/certificates\/ca.crt/" /etc/systemd/system/agent-protocol-forwarder.service
+elif [ -e /etc/certificates/tls.crt ] && [ -e /etc/certificates/tls.key ] && [ ! -e /etc/certificates/ca.crt ]; then
+    # Update systemd service file to add additional options
+    sed -i "/ExecStart/s/$/ \-cert\-file \/etc\/certificates\/tls.crt \-cert\-key \/etc\/certificates\/tls.key/" /etc/systemd/system/agent-protocol-forwarder.service
+fi
```

Comment:
How about using the `EnvironmentFile` directive in `agent-protocol-forwarder.service`, instead of directly modifying it?

`agent-protocol-forwarder.service` will looks like this.
```
[Service]
ExecStart=/usr/local/bin/agent-protocol-forwarder -kata-agent-namespace /run/netns/podns -kata-agent-socket /run/kata-containers/agent.sock $TLS_OPTIONS
EnvironmentFile=-/etc/default/agent-protocol-forwarder
```

`misc-settings.sh` will look like this.
```
if [ -e /etc/certificates/tls.crt ] && [ -e /etc/certificates/tls.key ] && [ -e /etc/certificates/ca.crt ]; then
   cat <<END >> /etc/default/agent-protocol-forwarder
TLS_OPTIONS=-cert-file /etc/certificates/tls.crt -cert-key /etc/certificates/tls.key -ca-cert-file /etc/certificates/ca.crt
END
elif
...
```

Reply:
I have used the above approach in the updated PR